### PR TITLE
feat(Anchor): Add icon to Anchor

### PR DESCRIPTION
**Pull request by LinKCoding**
### Overview
1. Adding an icon to the Anchor component. 
2. Applying logic from InlineIconButton's `getButtonContent()` function and refactoring that to work across Buttons, Badges, and Anchors. 
3. Added in some tests for Anchor (rendering purposes, can remove if it's unnecessary)
4. Updated figma link for icons (regular and mini)

### PR Checklist

- [x] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=60-3&m=dev
- [x] Related to JIRA ticket: [GM-704](https://codecademy.atlassian.net/browse/GM-704)
- [x] I have run this code to verify it works
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Go to `/?path=/docs/typography-anchor--anchor-with-icon#adding-an-icon` and check that the icon loads next to the Anchor (both left and right examples) 
2.  Go to `/?path=/docs/atoms-badge-badge--sample-icon#leading-icon` and check that the leading icon loads and looks centered
5. Go to `/?path=/docs/atoms-button--inline-icon#inline-icon` and check that the icons load properly
---  
6. Go to the Mono PR's Preview
7. Log in and navigate to the Profile (`profiles/<userID>`) page
8. Check the 'Latest Courses', if you don't have any courses, you should see: 
9. image
10.  If there are enrolled courses, check against the Streaks section, check that the streaks link renders an Icon 
11. image
 ---
12. Go to the Monolith PR's Preview
13. Go to the Signup page (`/register`) and check the bottom text's Terms & Conditions link and check that the icons render
14. image
15. NOTE: these visual tests will be removed before merging
16. ...
17. Profit!?

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37833)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Mono PR       | [Mono PR](https://github.com/codecademy-engineering/mono/pull/6483)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

[GM-704]: https://codecademy.atlassian.net/browse/GM-704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.2f60a5.0<br/>@codecademy/gamut-kit@0.6.411-alpha.2f60a5.0<br/>@codecademy/styleguide@66.21.2-alpha.2f60a5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6659e41acf4d1951f463efb0--gamut-preview.netlify.app>https://6659e41acf4d1951f463efb0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6659e41acf4d1951f463efb0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6659ea6a37abf65165ca1e78--gamut-preview.netlify.app>https://6659ea6a37abf65165ca1e78--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6659ea6a37abf65165ca1e78>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.c8d38c.0<br/>@codecademy/gamut-kit@0.6.411-alpha.c8d38c.0<br/>@codecademy/styleguide@66.21.2-alpha.c8d38c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6659eebe37abf65635ca1c67--gamut-preview.netlify.app>https://6659eebe37abf65635ca1c67--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6659eebe37abf65635ca1c67>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.f39777.0<br/>@codecademy/gamut-kit@0.6.411-alpha.f39777.0<br/>@codecademy/styleguide@66.21.2-alpha.f39777.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6659f1347ea6b0528d61456c--gamut-preview.netlify.app>https://6659f1347ea6b0528d61456c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6659f1347ea6b0528d61456c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.34236b.0<br/>@codecademy/gamut-kit@0.6.411-alpha.34236b.0<br/>@codecademy/styleguide@66.21.2-alpha.34236b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://665a10a4ea270e087ac148fe--gamut-preview.netlify.app>https://665a10a4ea270e087ac148fe--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/665a10a4ea270e087ac148fe>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.31b4b9.0<br/>@codecademy/gamut-kit@0.6.411-alpha.31b4b9.0<br/>@codecademy/styleguide@66.21.2-alpha.31b4b9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://665e17283107ad3fba81710a--gamut-preview.netlify.app>https://665e17283107ad3fba81710a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/665e17283107ad3fba81710a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.8af183.0<br/>@codecademy/gamut-kit@0.6.411-alpha.8af183.0<br/>@codecademy/styleguide@66.21.2-alpha.8af183.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://665e1ba7e990e740184500a4--gamut-preview.netlify.app>https://665e1ba7e990e740184500a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/665e1ba7e990e740184500a4>Deploy Logs</a></p>

File: packages/gamut/src/Anchor/index.tsx [hunk 1]
```diff
@@ -107,26 +108,54 @@ type AnchorBaseProps =
...
+      ...rest
+    },
+    ref
+  ) => {
+    const content = appendIconToContent({
+      iconPosition,
+      iconSize,
+      icon,
+      children,
+      center,
```
Comment (dreamwasp) [inline packages/gamut/src/Anchor/index.tsx:135]: i think this is where the not-centeredness is coming from, we prob always want the icon to be centered esp vertically

**Review by dreamwasp — CHANGES_REQUESTED** [review]
looking good so far! just a couple notes: 

the icons look a little uncentered:
image

you should be able to make the wrapper a flex container and style it like so to make it look like stacey's designs 
image
image

**Comment by LinKCoding** [comment]
> looking good so far! just a couple notes:
> 
> the icons look a little uncentered: image
> 
> you should be able to make the wrapper a flex container and style it like so to make it look like stacey's designs image image

Ah, I remember doing this, but because we need links to be inline I can't use `display:flex` unfortunately :\ 
I'm going to play around with `vertical-align` more but it doesn't look as centered as flex 👀 
Thanks!

**Comment by dreamwasp** [comment]
> > looking good so far! just a couple notes:
> > the icons look a little uncentered: image
> > you should be able to make the wrapper a flex container and style it like so to make it look like stacey's designs image image
> 
> Ah, I remember doing this, but because we need links to be inline I can't use `display:flex` unfortunately :\ I'm going to play around with `vertical-align` more but it doesn't look as centered as flex 👀 Thanks!

you could also try adding messing around with margin/padding `auto` or applying flex to just the container the icon is in!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.c20624.0<br/>@codecademy/gamut-kit@0.6.411-alpha.c20624.0<br/>@codecademy/styleguide@66.21.2-alpha.c20624.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6668c0a887ba4f092b8248eb--gamut-preview.netlify.app>https://6668c0a887ba4f092b8248eb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6668c0a887ba4f092b8248eb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.21.2-alpha.ff93eb.0<br/>@codecademy/gamut-kit@0.6.411-alpha.ff93eb.0<br/>@codecademy/styleguide@66.21.2-alpha.ff93eb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6669a1357604122ad5f225f7--gamut-preview.netlify.app>https://6669a1357604122ad5f225f7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6669a1357604122ad5f225f7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.2-alpha.50978b.0<br/>@codecademy/gamut-kit@0.6.414-alpha.50978b.0<br/>@codecademy/styleguide@66.24.1-alpha.50978b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6669acac6363602cd50d76b6--gamut-preview.netlify.app>https://6669acac6363602cd50d76b6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6669acac6363602cd50d76b6>Deploy Logs</a></p>

File: packages/styleguide/stories/Typography/Anchor.examples.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  MiniArrowRightIcon,
+  MiniInfoOutlineIcon,
+  MiniStarIcon,
+} from '@codecademy/gamut-icons';
+import { Background } from '@codecademy/gamut-styles';
+import startCase from 'lodash/startCase';
+
+const variants = ['inline', 'interface', 'standard'] as const;
+
+export const VariantsExample = ({ useIcon }: { useIcon: boolean }) => {
```
Comment (LinKCoding) [inline packages/styleguide/stories/Typography/Anchor.examples.tsx:19]: opting only to use a boolean to render the icon or not for simplicity's sake.

File: packages/styleguide/stories/Typography/Anchor.examples.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+              </Anchor>
+            </Box>
+          ))}
+        </Background>
+      </Column>
+    </LayoutGrid>
+  );
+};
+
+export const InlineAnchorWithIconAndText = ({ args }: { args: any }) => {
```
Comment (LinKCoding) [inline packages/styleguide/stories/Typography/Anchor.examples.tsx:74]: also keeping this as `any` for simplicity's sake.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.2-alpha.882f34.0<br/>@codecademy/gamut-kit@0.6.414-alpha.882f34.0<br/>@codecademy/styleguide@66.24.1-alpha.882f34.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6669d6185d9685007af2337c--gamut-preview.netlify.app>https://6669d6185d9685007af2337c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6669d6185d9685007af2337c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.2-alpha.1179ba.0<br/>@codecademy/gamut-kit@0.6.414-alpha.1179ba.0<br/>@codecademy/styleguide@66.24.1-alpha.1179ba.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6669f265ac27de1d0011db3f--gamut-preview.netlify.app>https://6669f265ac27de1d0011db3f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6669f265ac27de1d0011db3f>Deploy Logs</a></p>

File: packages/styleguide/stories/Typography/Anchor.examples.tsx (hunk 3 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Typography/Anchor.examples.tsx:74]: i tested this locally and it worked for me! you may need to tweak the way you're passing the args, only tested for type safety herer

`export const InlineAnchorWithIconAndText: React.FC<AnchorProps> = (args) => {`

Comment (LinKCoding) [reply packages/styleguide/stories/Typography/Anchor.examples.tsx:74]: Does that mean i need to do more than replace my line of code for what you wrote? 😂

Comment (dreamwasp) [reply packages/styleguide/stories/Typography/Anchor.examples.tsx:74]: maybe just change the way you're passing args in the story!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.2-alpha.f51443.0<br/>@codecademy/gamut-kit@0.6.414-alpha.f51443.0<br/>@codecademy/styleguide@66.24.1-alpha.f51443.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://666b7335e5d72bc8accf7e42--gamut-preview.netlify.app>https://666b7335e5d72bc8accf7e42--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/666b7335e5d72bc8accf7e42>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66758b396a08c692f4eb4c1a--gamut-preview.netlify.app>https://66758b396a08c692f4eb4c1a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66758b396a08c692f4eb4c1a>Deploy Logs</a></p>

File: packages/gamut/src/Anchor/index.tsx [hunk 4]
```diff
@@ -107,26 +108,59 @@ type AnchorBaseProps =
...
+      iconSize = 12,
+      iconAndTextGap = 8,
+      isInlineIcon = true,
+      variant = 'inline',
+      ...rest
+    },
+    ref
+  ) => {
+    // At the moment, inconOffset is hardcoded for font-size 16px, but if needed, we can expose this for developers
+    iconOffset = variant === 'inline' ? 2 : 4;
```
Comment (dreamwasp) [inline packages/gamut/src/Anchor/index.tsx:132]: can you make this scalable using `em`s? you might need to cast it as an any but i'm wondering if that would fix our font iscaling ssues a bit! (`em`s are always relative to their parent components ✨ )

**Review by dreamwasp — COMMENTED** [review]
this looks great!! once the story is cleaned up i'm happy to ✅  even if the offset is left as is 🔥

File: packages/gamut/src/helpers/appendIconToContent.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  iconPosition,
+  iconSize = 12,
+  isInlineIcon = false,
+}: AppendedIconProps) => {
+  if (!Icon) return <>{children}</>;
+
+  const iconSpacing = iconPosition === 'left' ? 'mr' : 'ml';
+  const iconPositioning = iconPosition === 'left' ? 0 : 1;
+
+  const iconProps = {
```
Comment (LinKCoding) [inline packages/gamut/src/helpers/appendIconToContent.tsx:43]: This is to check if `iconOffset` is assigned a number (i.e. an override) -- but if it's just `undefined`, then 2 and 4 should work respectively.

**Comment by LinKCoding** [comment]
pixels vs em comparison. 
the following SS show the diff between pixels and em offset implementations 
The top paragraph in the SS changes (and is noted in the text under the SS) the bottom text stays as 16px. 

image
Just to note, the pixel implementation works well even when on larger font sizes. 

image
^ it looks off tho when font is smaller than 16px (it's 14 in the SS) 

image
^ this is with em, top text being 14px, it looks a bit more centered. 

image
^ and this is size 22px with em. 

Gonna go ahead and make the change to em -- but @dreamwasp just to note, I added in a helper to convert px to em 
If there's a preferred library or a different implementation please lmk 🙏 
Thanks~

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.6-alpha.a0bee8.0<br/>@codecademy/gamut-kit@0.6.418-alpha.a0bee8.0<br/>@codecademy/styleguide@66.26.1-alpha.a0bee8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6675e1be769adadf7b58df16--gamut-preview.netlify.app>https://6675e1be769adadf7b58df16--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6675e1be769adadf7b58df16>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
this is wonderful, and thank you for being so thorough 🙏

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.22.7-alpha.95a2b7.0<br/>@codecademy/gamut-kit@0.6.419-alpha.95a2b7.0<br/>@codecademy/styleguide@66.26.2-alpha.95a2b7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://667ae30b9da3d01f18f5c1fa--gamut-preview.netlify.app>https://667ae30b9da3d01f18f5c1fa--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/667ae30b9da3d01f18f5c1fa>Deploy Logs</a></p>

**Pull request merged**